zkvm/src/vm.rs

```diff
@@ -331,6 +331,38 @@ where
         Ok(())
     }
 
+    fn unblind(&mut self) -> Result<(), VMError> {
+        // Pop expression `expr`
+        let expr = self.pop_item()?.to_expression()?;
+
+        // Pop commitment `V`
+        let v_commitment = self.pop_item()?.to_data()?.to_commitment()?;
+        let v_point = v_commitment.to_point();
+
+        // Pop scalar `v`
+        let data = self.pop_item()?.to_data()?.to_bytes();
+        let v_scalar = SliceReader::parse(&data, |r| r.read_scalar())?;
```

Comment:
We already have `Data::Scalar` case, so it would make sense to do:

```rust
        let scalar_witness = self.pop_item()?.to_data()?.to_scalar()?;
        let v_scalar = scalar_witness.to_scalar();
```

where `Data::to_scalar` would either parse Data::Opaque, or return the contents of Data::Scalar (which is ScalarWitness, so we'd need to convert _that_ to raw scalar again).

And we do that, we should use this scalar witness in `fn r#const` too.